os/arch/arm/Kconfig

```diff
@@ -288,6 +288,16 @@ config ARCH_DABORTSTACK
 		will be used during data abort handling. Recommended data abort stack
 		size is 1K.
 
+config REG_STACK_OVERFLOW_PROTECTION
+	bool "REG Stack Overflow Protection"
+	depends on BUILD_PROTECTED
```

Comment:
Why does this option depend on protected build? I think this stack limit check applies to flat build also.

Reply:
The below config `MPU_STACK_OVERFLOW_PROTECTION` also need not have dependency with protected build. Please check this also.

Reply:
removed dependency of reg_stack_overflow_protection on protected build..
for mpu_stack will create other PR